### PR TITLE
feat(checkout): Add custom checkout command

### PR DESCRIPTION
**Pull request by nuclearcat**
Add command to trigger custom checkout and run all or selected test on this checkout.

### CODE REVIEW

File: kci-dev/subcommands/checkout.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    is_flag=True,
+)
+# jobfilter is a list, might be one or more jobs
+@click.option(
+    "--jobfilter",
+    help="Job filter to trigger",
+    multiple=True,
+)
+@click.pass_context
+def checkout(ctx, giturl, branch, commit, jobfilter, tipoftree):
```
Comment (aliceinwire) [inline kci-dev/subcommands/checkout.py:95]: checkout is not the goal of this command. the goal is to trigger a test of the custom kernel tree. checkout is only a process  of getting the custom kernel tree.

**Review by aliceinwire — APPROVED** [review]

**Comment by aliceinwire** [comment]
documentation is missing

**Comment by nuclearcat** [comment]
Yes, still in progress, will add as separate commit

**Comment by aliceinwire** [comment]
I'm not sure if this should be merged with the commit subcommand

**Comment by aliceinwire** [comment]
> What are other options? I'm open to suggestions.
> 
> As next plans, in separate PR i am planning to add --watch option to this checkout option, so this will be key tool to automated bisection, key option. So for example: kci-dev bisect badcommit goodcommit kbuildname testsuite [testname] will run git bisect, obtain first suggested by it commit and start
> 
> ```
> kci-dev checkout (tree and branch options and jobfilter) --commit firstcommit --watch testname/testsuite
> ```
> 
> Then it will track progress, how tree being built and etc. If result fail -it will give "git bisect bad" (and if good - opposite) and test next commit, until we find offending commit. P.S. Also as Pawel suggested, it might be that some commits are not building, so it will be git bisect skip in such cases.

I see! I'm ok with merging the checkout command. I was just thinking how much was different from the commit command but this partially answered the question and I'm fine with having this.

**Comment by aliceinwire** [comment]
checkout.md is empty i suppose we will write with the progress of the command features

**Comment by nuclearcat** [comment]
> checkout.md is empty I suppose we will write with the progress of the command features
One minute, seems deleted content by mistake.

**Pull request merged**